modules/ctl/src/main/scala/free/interp.scala

```diff
@@ -0,0 +1,119 @@
+package gem.ctl
+package free
+
+import scalaz._, Scalaz._
+import scalaz.effect._
+
+import gem.ctl.low.io._
+import gem.ctl.free.ctl._
+
+/** An interpreter for the control language. */
+object interpreter {
+
+  /** Configuration for the interpreter. */
+  trait Config {
+    def verbose: Boolean
+    def server:  Server
+  }
+
+  case class InterpreterState(indentation: Int, machineHostCache: Map[Host.Machine, String]) {
+    def indent  = copy(indentation = indentation + 1)
+    def outdent = copy(indentation = indentation - 1)
+  }
+  object InterpreterState {
+    val initial = InterpreterState(0, Map.empty)
+  }
+
+  /**
+   * Construct an interpreter of `CtlIO` given a `Config` and an `IORef` for the initial state. We
+   * carry our state in an `IORef` because it needs to be visible from multiple threads.
+   */
+  def interpreter(c: Config, state: IORef[InterpreterState]) = λ[CtlOp ~> EitherT[IO, Int, ?]] {
```

Comment:
This is a new thing in kind-projector that makes it easier to construct natural transformations and things like it.

```scala
λ[F ~> G]{ ... }
```

Is more or less the same thing as 

```scala
new (F ~> G) {
  def impl[A]: F[A] => G[A] = ...
  def apply[A](fa: F[A]): G[A] = impl[A](fa)
}
```

There's more info in the kind-projector doc. This is new but already pretty standard now. It's used in cats and other libs.